source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1054 @@
+          prm.declare_entry("Number of horizons", "1",
+                            Patterns::Integer(),
+                            "Number of horizons to track and visualize in FastScape..");
+          prm.declare_entry("Y extent in 2d", "100000",
+                            Patterns::Double(),
+                            "Y extent when used with 2D");
+
+
+          prm.enter_subsection ("Boundary conditions");
+          {
+            prm.declare_entry ("Bottom", "1",
+                               Patterns::Integer (0, 1),
+                               "Bottom boundary condition, where 1 is fixed and 0 is reflective.");
+            prm.declare_entry ("Right", "1",
+                               Patterns::Integer (0, 1),
+                               "Right boundary condition, where 1 is fixed and 0 is reflective.");
+            prm.declare_entry ("Top", "1",
+                               Patterns::Integer (0, 1),
+                               "Top boundary condition, where 1 is fixed and 0 is reflective.");
+            prm.declare_entry ("Left", "1",
+                               Patterns::Integer (0, 1),
+                               "Left boundary condition, where 1 is fixed and 0 is reflective.");
+            prm.declare_entry("Left mass flux", "0",
+                              Patterns::Double(),
+                              "Flux per unit length through left boundary (m^2/yr)");
+            prm.declare_entry("Right mass flux", "0",
+                              Patterns::Double(),
+                              "Flux per unit length through right boundary (m^2/yr)");
+            prm.declare_entry("Top mass flux", "0",
+                              Patterns::Double(),
+                              "Flux per unit length through top boundary (m^2/yr)");
+            prm.declare_entry("Bottom mass flux", "0",
+                              Patterns::Double(),
+                              "Flux per unit length through bottom boundary (m^2/yr)");
+          }
+          prm.leave_subsection();
+
+          prm.enter_subsection ("Erosional parameters");
+          {
+            prm.declare_entry("Drainage area exponent", "0.4",
+                              Patterns::Double(),
+                              "Exponent for drainage area.");
+            prm.declare_entry("Slope exponent", "1",
+                              Patterns::Double(),
+                              "The  slope  exponent  for  SPL (n).  Generally  m/n  should  equal  approximately 0.4");
+            prm.declare_entry("Multi-direction slope exponent", "1",
+                              Patterns::Double(),
+                              "Exponent to determine the distribution from the SPL to neighbor nodes, with"
+                              "10 being steepest decent and 1 being more varied.");
+            prm.declare_entry("Bedrock deposition coefficient", "-1",
```

Comment:
do these negative default values make sense?

Reply:
FastScape takes values < 0 to imply that the value isn't used, although maybe using zero would work as well and be less confusing.